### ISSUE TITLE
pkgcruft FilesUnused false positive if file is consumed via "non-standard" variable

### ISSUE THREAD

**Issue by Flowdalic** [pkgcraft/pkgcraft#237]
For example, pkgcruft reports:

```
x11-wm/stumpwm
  FilesUnused: 70stumpwm-gentoo.el
```

however `stumpwm-22.05_p20220818.ebuild` contains
```
SITEFILE=70${PN}-gentoo.el
```

Maybe pkgcruft should not consider the file used if the filename is contained in a (global) variable

**Comment by radhermit** [comment]
Does that file actually get installed? The check already short circuits if any inherited eclass contains a FILESDIR reference for exactly this reason.

I specifically remember checking this ebuild (since it's near the end of the alphabet and thus was easy to find when experimenting with the check) and not finding any FILESDIR usage in its inheritance tree because it inherits elisp-common and not elisp which is the eclass that actually installs the SITEFILE references automatically. IIRC, elisp-common requires explicitly calling something like `elisp-site-file-install "${FILESDIR}"/${SITEFILE}` which you can find in all other elisp-common inherit ebuilds across the tree, e.g. x11-wm/ratpoison.

TLDR, I don't think this is a false positive.

**Comment by radhermit** [comment]
Closing since I just confirmed my previous statement by test installing ratpoison and stumpwm.

For ratpoison using `find . -name *.el` in the image directory returns `/usr/share/emacs/site-lisp/ratpoison.el` while stumpwm returns nothing because it erroneously isn't installed that file via `elisp-site-file-install` as mentioned previously.

That being said, there are probably other types of false positives (although I did try to skip the major ones) so please report any you want to see fixed.

**Comment by Flowdalic** [comment]
I think you are right. I intuitively assumed that pkgcruft dirs the wrong thing, when it seems to do the right thing. What a pleasant surprise. 🙃
Sorry for the noise
